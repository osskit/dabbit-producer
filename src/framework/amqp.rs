use lapin::{
    BasicProperties, Connection,
    ConnectionProperties, Error
};

async fn connect(amqp_addr: &str) -> Result<Connection, Error> {
    Connection::connect(
        &amqp_addr,
        ConnectionProperties::default(),
    )
    .await
}

async fn create_channel(connection: Connection) -> Result<(), Error>{
    let channel_a = connection.create_channel().await?;

    let queue = channel_a
        .queue_declare(
            "hello",
            QueueDeclareOptions::default(),
            FieldTable::default(),
        )
        .await?;

    let payload = b"Hello world!";

    loop {
        let confirm = channel_a
            .basic_publish(
                "",
                "hello",
                BasicPublishOptions::default(),
                payload,
                BasicProperties::default(),
            )
            .await?
            .await?;
        assert_eq!(confirm, Confirmation::NotRequested);
    }
}