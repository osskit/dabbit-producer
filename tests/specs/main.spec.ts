import {init, close, get} from '../amqp';

describe('main', () => {
    beforeAll(async () => {
        await init();
    });

    afterAll(async () => {
        await close();
    });

    it('readiness', async () => {
        const {ok} = await fetch('http://localhost:3000/ready');

        expect(ok).toBeTruthy();
    });

    it('produce', async () => {
        const {ok} = await fetch('http://localhost:3000/produce', { 
            method: 'POST', 
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({})}
         );

        expect(ok).toBeTruthy();

        const message = await get();

        expect(message).toMatchSnapshot();
    });
});