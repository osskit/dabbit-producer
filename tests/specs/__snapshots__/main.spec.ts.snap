// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main produce 1`] = `null`;
